lib/datadog/open_feature/binding/configuration.rb

```diff
@@ -0,0 +1,292 @@
+# frozen_string_literal: true
+
+module Datadog
+  module OpenFeature
+    module Binding
+      # Variation types supported by UFC
+      module VariationType
+        STRING = 'STRING'
+        INTEGER = 'INTEGER'
+        NUMERIC = 'NUMERIC'
+        BOOLEAN = 'BOOLEAN'
+        JSON = 'JSON'
+      end
+
+      # Condition operators for rule evaluation
+      module ConditionOperator
+        MATCHES = 'MATCHES'
+        NOT_MATCHES = 'NOT_MATCHES'
+        GTE = 'GTE'
+        GT = 'GT'
+        LTE = 'LTE'
+        LT = 'LT'
+        ONE_OF = 'ONE_OF'
+        NOT_ONE_OF = 'NOT_ONE_OF'
+        IS_NULL = 'IS_NULL'
+      end
+
+      # Assignment reasons returned in ResolutionDetails
+      module AssignmentReason
+        TARGETING_MATCH = 'TARGETING_MATCH'
+        SPLIT = 'SPLIT'
+        STATIC = 'STATIC'
+      end
+
+      # Represents a feature flag configuration
+      class Flag
+        attr_reader :key, :enabled, :variation_type, :variations, :allocations
+
+        def initialize(key:, enabled:, variation_type:, variations:, allocations:)
+          @key = key
+          @enabled = enabled
+          @variation_type = variation_type
+          @variations = variations || {}
+          @allocations = allocations || []
+        end
+
+        def self.from_json(key, flag_data)
+          new(
+            key: key,
+            enabled: flag_data['enabled'] || false,
+            variation_type: flag_data['variationType'],
```

Comment:
if we want to ensure `variation_type` presence, it's better to use `fetch` here as well

Reply:
https://github.com/DataDog/dd-trace-rb/pull/5022/commits/b85c6e4f9309c36cca924411a7c7fa9ff83889ce